src/Pretzel.Logic/Commands/BaseParameters.cs

```diff
@@ -1,67 +1,37 @@
-using NDesk.Options;
+using Pretzel.Logic.Extensibility;
 using System;
 using System.Collections.Generic;
+using System.CommandLine;
 using System.Composition;
-using System.IO.Abstractions;
 using System.Linq;
 
 namespace Pretzel.Logic.Commands
 {
-    public sealed class BaseParameters
+    public abstract class BaseParameters : ICommandParameters, ICommandParametersExtendable
     {
-        public OptionSet Options { get; }
+        [ImportMany]
+        public ExportFactory<IHaveCommandLineArgs, CommandArgumentsExtentionAttribute>[] ArgumentExtenders { get; set; }
 
-        public string CommandName { get; }
+        [Export]
+        public IList<Option> Options { get; set; }
 
-        public bool Help { get; private set; }
-
-        public bool Debug { get; private set; }
-
-        public bool Safe { get; private set; }
-
-        public string Path { get; private set; }
-
-        public IFileSystem FileSystem { get; private set; }
-
-        public List<string> CommandArgs { get; private set; }
-
-        private BaseParameters(string[] arguments, IFileSystem fileSystem)
+        [OnImportsSatisfied]
+        internal void OnImportsSatisfied()
         {
-            Options = new OptionSet
-            {
-                { "help", "Display help mode", p => Help = true },
-                { "debug", "Enable debugging", p => Debug = true },
-                { "safe", "Disable custom plugins", v => Safe = true },
-                { "d|directory=", "[Obsolete, use --source instead] The path to site directory", p => Path = p },
-                { "s|source=", "The path to the source site (default current directory)", p => Path = p }
-            };
-
-            FileSystem = fileSystem;
+            var options = new List<Option>();
+            Options = options;
+            WithOptions(options);
 
-            CommandName = arguments.Take(1).FirstOrDefault();
-
-            CommandArgs = Options.Parse(arguments.Skip(1));
-
-            SetPath(CommandArgs.FirstOrDefault());
+            foreach (var factory in this.GetCommandExtentions())
+            {
+                factory.CreateExport().Value.UpdateOptions(Options);
+            }
         }
 
-        public static BaseParameters Parse(string[] arguments, IFileSystem fileSystem)
-        {
-            return new BaseParameters(arguments, fileSystem);
-        }
+        protected abstract void WithOptions(List<Option> options);
```

Comment:
I'd prefer that this function doesn't pass the options list but only get the specific options and add them to the main list.
Or is it necessary for something that I didn't see (yet)?

Reply:
Agree.